core-java-modules/core-java-string-operations-3/src/test/java/com/baeldung/versioncomparison/VersionComparisonUnitTest.java

```diff
@@ -0,0 +1,136 @@
+package com.baeldung.versioncomparison;
+
+import org.junit.Test;
+
+import com.fasterxml.jackson.core.Version;
+import com.vdurmont.semver4j.Semver;
+import com.vdurmont.semver4j.Semver.VersionDiff;
+
+import org.apache.maven.artifact.versioning.ComparableVersion;
+import org.gradle.util.VersionNumber;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class VersionComparisonUnitTest {
+    
+    @Test
+    public void givenVersionStrings_whenUsingMavenArtifact_thenCompareVersions() {
+        ComparableVersion version1_1 = new ComparableVersion("1.1");
+        ComparableVersion version1_2 = new ComparableVersion("1.2");
+        ComparableVersion version1_3 = new ComparableVersion("1.3");
+
+        assertTrue(version1_1.compareTo(version1_2) < 0);
+        assertTrue(version1_3.compareTo(version1_2) > 0);
+        
+        ComparableVersion version1_1_0 = new ComparableVersion("1.1.0"); 
+        assertEquals(version1_1.compareTo(version1_1_0), 0);
+
+        ComparableVersion version1_1_alpha = new ComparableVersion("1.1-alpha");
+        assertTrue(version1_1.compareTo(version1_1_alpha) > 0);
```

Comment:
swap order of arguments so that you get assertTrue(expected, actual)

Reply:
I think it is relevant for assertEquals

Reply:
I think, it is only relevant for assertEquals